src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -201,13 +201,31 @@ def check_lazy(*args: Any, **kwargs: Any) -> Any:
                     returned_value = Index(returned_value, convert_to_lazy=False)
                 # Some methods also return a tuple with a pandas Index, so convert the tuple's first item to a modin Index
                 # Examples of this are `_get_indexer_strict` and `sort_values`
+
+                # We want to return a native pandas series for value counts because our loc/iloc functionality relies on it
+                # If we returned a snowpark pandas series, this would result in querying data many times
+                elif func_name == "value_counts":
```

Comment:
if we don't do this here, what happens?

Reply:
Some of the loc methods go to 18 queries because we query data each time, see [here](https://github.com/snowflakedb/snowpark-python/blob/68a2051b8fa5a5b8f436487cff2c3e6610d8eaaa/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py#L778) for the details